packages/sdk/src/protocols/base_protocol.ts

```diff
@@ -42,22 +42,17 @@ export class BaseProtocolSDK implements IBaseProtocolSDK {
   /**
    * Disconnects from a peer and tries to find a new one to replace it.
    * @param peerToDisconnect The peer to disconnect from.
+   * @returns The new peer that was found and connected to.
    */
-  public async renewPeer(peerToDisconnect: PeerId): Promise<void> {
+  public async renewPeer(peerToDisconnect: PeerId): Promise<Peer> {
     this.log.info(`Renewing peer ${peerToDisconnect}`);
-    try {
-      await this.connectionManager.dropConnection(peerToDisconnect);
-      this.peers = this.peers.filter((peer) => peer.id !== peerToDisconnect);
-      this.log.info(
-        `Peer ${peerToDisconnect} disconnected and removed from the peer list`
-      );
+    await this.connectionManager.dropConnection(peerToDisconnect);
+    this.peers = this.peers.filter((peer) => peer.id !== peerToDisconnect);
+    this.log.info(
+      `Peer ${peerToDisconnect} disconnected and removed from the peer list`
+    );
 
-      await this.findAndAddPeers(1);
-    } catch (error) {
-      this.log.info(
-        "Peer renewal failed, relying on the interval to find a new peer"
-      );
-    }
+    return (await this.findAndAddPeers(1))[0];
```

Comment:
nit: let's not refer by index without any guards as we have some problems before

Reply:
```suggestion
    const peer = (await this.findAndAddPeers(1))[0];
    
    if (!peer) throw;
    
    return peer;
```